src/Billing/Controllers/StripeController.cs

```diff
@@ -802,12 +803,14 @@ private async Task<Subscription> VerifyCorrectTaxRateForCharge(Invoice invoice,
         private static bool IsSponsoredSubscription(Subscription subscription) =>
             StaticStore.SponsoredPlans.Any(p => p.StripePlanId == subscription.Id);
 
-        private async Task HandlePaymentFailed(Invoice invoice, Subscription subscription)
+        private async Task HandlePaymentFailed(Invoice invoice)
         {
             if (!invoice.Paid && invoice.AttemptCount > 1 && UnpaidAutoChargeInvoiceForSubscriptionCycle(invoice))
             {
+                var subscriptionService = new SubscriptionService();
+                var subscription = await subscriptionService.GetAsync(invoice.SubscriptionId);
                 // attempt count 4 = 11 days after initial failure
-                if (invoice.AttemptCount > 3 && subscription.Items.Any(i => i.Plan.Id == PremiumPlanId))
+                if (invoice.AttemptCount > 3 && subscription.Items.Any(i => i.Price.Id == PremiumPlanId || i.Price.Id == PremiumPlanIdAppStore))
```

Comment:
Do we need to swap this same fix on line 422-423?
`if (subscription.Items.Any(i => i.Plan.Id == PremiumPlanId))` to use Price.Id instead of Plan.Id?

Reply:
I don't know. I copied that original line FROM 422 (assuming it worked as advertised) but it does not. We need to revisit that separately and make sure it's working, but there haven't been any reported issues so I'm not sure what's going on there..